src/btree/bt_split.c

```diff
@@ -1476,6 +1476,11 @@ __split_multi_inmem(WT_SESSION_IMPL *session, WT_PAGE *orig, WT_MULTI *multi, WT
             /* Search the page. */
             WT_ERR(__wt_row_search(&cbt, key, true, ref, true, NULL));
 
+            /* If required, remove the existing prepared update restored from the disk. */
+            if (supd->restore && supd->ins == NULL && prepare &&
+              !F_ISSET(S2C(session), WT_CONN_IN_MEMORY))
```

Comment:
Why exclude in memory? In memory also does update restore.

Reply:
We don't call `__wt_page_inmem_prepare()` for in-memory case:

```
   /*
     * In-memory databases restore non-obsolete updates directly in this function, don't call the
     * underlying page functions to do it.
     */
    if (prepare && !F_ISSET(S2C(session), WT_CONN_IN_MEMORY))
        WT_RET(__wt_page_inmem_prepare(session, ref));
```